### PR TITLE
refactor(core): remove circular reference from cloud provider bake handlers and its configuration classes identified during upgrade to spring boot 2.6.x

### PR DESCRIPTION
**Pull request by j-sandy**
While upgrading spring boot 2.6.15 and spring cloud 2021.0.8, encountered below errors in start up tests of rosco-core module during test execution and 5 tests failed with similar error:
```
WARN 142936 --- [    Test worker] o.s.w.c.s.GenericWebApplicationContext   : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'AWSBakeHandler': Unsatisfied dependency expressed through field 'awsBakeryDefaults'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.netflix.spinnaker.rosco.providers.aws.config.RoscoAWSConfiguration': Unsatisfied dependency expressed through field 'awsBakeHandler'; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'AWSBakeHandler': Requested bean is currently in creation: Is there an unresolvable circular reference?
INFO 142936 --- [    Test worker] ConditionEvaluationReportLoggingListener :

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
ERROR 142936 --- [    Test worker] o.s.b.d.LoggingFailureAnalysisReporter   :

***************************
APPLICATION FAILED TO START
***************************

Description:

The dependencies of some of the beans in the application context form a cycle:

┌─────┐
|  AWSBakeHandler (field private com.netflix.spinnaker.rosco.providers.aws.config.RoscoAWSConfiguration$AWSBakeryDefaults com.netflix.spinnaker.rosco.providers.aws.AWSBakeHandler.awsBakeryDefaults)
↑     ↓
|  com.netflix.spinnaker.rosco.providers.aws.config.RoscoAWSConfiguration (field private com.netflix.spinnaker.rosco.providers.aws.AWSBakeHandler com.netflix.spinnaker.rosco.providers.aws.config.RoscoAWSConfiguration.awsBakeHandler)
└─────┘

Action:

Relying upon circular references is discouraged and they are prohibited by default. Update your application to remove the dependency cycle between beans. As a last resort, it may be possible to break the cycle automatically by setting spring.main.allow-circular-references to true.

ERROR 142936 --- [    Test worker] o.s.test.context.TestContextManager      : Caught exception while allowing TestExecutionListener [org.springframework.test.context.web.ServletTestExecutionListener@1d1cfe4] to prepare test instance [com.netflix.spinnaker.rosco.providers.aws.AWSStartupTest@2a5782d7]

```

```
WARN 142936 --- [    Test worker] o.s.w.c.s.GenericWebApplicationContext   : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'azureBakeHandler': Unsatisfied dependency expressed through field 'azureBakeryDefaults'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.netflix.spinnaker.rosco.providers.azure.config.RoscoAzureConfiguration': Unsatisfied dependency expressed through field 'azureBakeHandler'; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'azureBakeHandler': Requested bean is currently in creation: Is there an unresolvable circular reference?
INFO 142936 --- [    Test worker] ConditionEvaluationReportLoggingListener :

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2024-02-27 13:33:50.968 ERROR 142936 --- [    Test worker] o.s.b.d.LoggingFailureAnalysisReporter   :

***************************
APPLICATION FAILED TO START
***************************

Description:

The dependencies of some of the beans in the application context form a cycle:

┌─────┐
|  azureBakeHandler (field private com.netflix.spinnaker.rosco.providers.azure.config.RoscoAzureConfiguration$AzureBakeryDefaults com.netflix.spinnaker.rosco.providers.azure.AzureBakeHandler.azureBakeryDefaults)
↑     ↓
|  com.netflix.spinnaker.rosco.providers.azure.config.RoscoAzureConfiguration (field private com.netflix.spinnaker.rosco.providers.azure.AzureBakeHandler com.netflix.spinnaker.rosco.providers.azure.config.RoscoAzureConfiguration.azureBakeHandler)
└─────┘
```

```
WARN 142936 --- [    Test worker] o.s.w.c.s.GenericWebApplicationContext   : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'dockerBakeHandler': Unsatisfied dependency expressed through field 'dockerBakeryDefaults'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.netflix.spinnaker.rosco.providers.docker.config.RoscoDockerConfiguration': Unsatisfied dependency expressed through field 'dockerBakeHandler'; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'dockerBakeHandler': Requested bean is currently in creation: Is there an unresolvable circular reference?
INFO 142936 --- [    Test worker] ConditionEvaluationReportLoggingListener :

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
ERROR 142936 --- [    Test worker] o.s.b.d.LoggingFailureAnalysisReporter   :

***************************
APPLICATION FAILED TO START
***************************

Description:

The dependencies of some of the beans in the application context form a cycle:

┌─────┐
|  dockerBakeHandler (field private com.netflix.spinnaker.rosco.providers.docker.config.RoscoDockerConfiguration$DockerBakeryDefaults com.netflix.spinnaker.rosco.providers.docker.DockerBakeHandler.dockerBakeryDefaults)
↑     ↓
|  com.netflix.spinnaker.rosco.providers.docker.config.RoscoDockerConfiguration (field private com.netflix.spinnaker.rosco.providers.docker.DockerBakeHandler com.netflix.spinnaker.rosco.providers.docker.config.RoscoDockerConfiguration.dockerBakeHandler)
└─────┘

```

```
WARN 142936 --- [    Test worker] o.s.w.c.s.GenericWebApplicationContext   : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'GCEBakeHandler': Unsatisfied dependency expressed through field 'gceBakeryDefaults'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.netflix.spinnaker.rosco.providers.google.config.RoscoGoogleConfiguration': Unsatisfied dependency expressed through field 'gceBakeHandler'; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'GCEBakeHandler': Requested bean is currently in creation: Is there an unresolvable circular reference?
INFO 142936 --- [    Test worker] ConditionEvaluationReportLoggingListener :

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
ERROR 142936 --- [    Test worker] o.s.b.d.LoggingFailureAnalysisReporter   :

***************************
APPLICATION FAILED TO START
***************************

Description:

The dependencies of some of the beans in the application context form a cycle:

┌─────┐
|  GCEBakeHandler (field private com.netflix.spinnaker.rosco.providers.google.config.RoscoGoogleConfiguration$GCEBakeryDefaults com.netflix.spinnaker.rosco.providers.google.GCEBakeHandler.gceBakeryDefaults)
↑     ↓
|  com.netflix.spinnaker.rosco.providers.google.config.RoscoGoogleConfiguration (field private com.netflix.spinnaker.rosco.providers.google.GCEBakeHandler com.netflix.spinnaker.rosco.providers.google.config.RoscoGoogleConfiguration.gceBakeHandler)
└─────┘

```

```
WARN 142936 --- [    Test worker] o.s.w.c.s.GenericWebApplicationContext   : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'OCIBakeHandler': Unsatisfied dependency expressed through field 'oracleBakeryDefaults'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.netflix.spinnaker.rosco.providers.oracle.config.RoscoOracleConfiguration': Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'OCIBakeHandler': Requested bean is currently in creation: Is there an unresolvable circular reference?
INFO 142936 --- [    Test worker] ConditionEvaluationReportLoggingListener :

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
ERROR 142936 --- [    Test worker] o.s.b.d.LoggingFailureAnalysisReporter   :

***************************
APPLICATION FAILED TO START
***************************

Description:

The dependencies of some of the beans in the application context form a cycle:

┌─────┐
|  OCIBakeHandler (field com.netflix.spinnaker.rosco.providers.oracle.config.OracleBakeryDefaults com.netflix.spinnaker.rosco.providers.oracle.OCIBakeHandler.oracleBakeryDefaults)
↑     ↓
|  com.netflix.spinnaker.rosco.providers.oracle.config.RoscoOracleConfiguration
└─────┘

```
The Root cause is circular reference between bake handler and respective rosco configuration classes of cloud providers. To resolve circular reference issue, separated the bean creation from rosco configuration classes to a new file for AWS, Azure, Google, Docker and Oracle cloud providers.

### CODE REVIEW

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/aws/config/AWSBakeryDefaultsBean.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2024 Netflix, Inc.
```
Comment (dbyron-sf) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/aws/config/AWSBakeryDefaultsBean.java:2]:
```suggestion
 * Copyright 2024 OpsMx, Inc.
```

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/azure/config/AzureBakeryConfigurationBeans.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 Netflix, Inc.
```
Comment (dbyron-sf) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/azure/config/AzureBakeryConfigurationBeans.java:2]:
```suggestion
 * Copyright 2024 OpsMx, Inc.
```

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/docker/config/DockerBakeryDefaultsBean.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 Netflix, Inc.
```
Comment (dbyron-sf) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/docker/config/DockerBakeryDefaultsBean.java:2]:
```suggestion
 * Copyright 2024 OpsMx, Inc.
```

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/google/config/GCEBakeryConfigurationBeans.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Netflix, Inc.
```
Comment (dbyron-sf) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/google/config/GCEBakeryConfigurationBeans.java:2]:
```suggestion
 * Copyright 2024 OpsMx, Inc.
```

File: rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/oracle/config/OracleBakeryConfigurationBeans.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 Netflix, Inc.
```
Comment (dbyron-sf) [inline rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/oracle/config/OracleBakeryConfigurationBeans.java:2]:
```suggestion
 * Copyright 2024 OpsMx, Inc.
```

**Comment by j-sandy** [comment]
@Mergifyio update

**Pull request closed without merging**